### PR TITLE
[app_docs] add `hclFormat` func

### PR DESCRIPTION
**Pull request by thiskevinwang**
### What

This is a poc/proposal to add a `hclFormat` func, along with a `gen/integrations-hcl` `Makefile` target.

https://user-images.githubusercontent.com/26389321/208122611-bb22ee72-9479-48d2-b416-37aacc3594dd.mp4

### Why

This automates the creation of a few files, that power the **Integrations** project. (See: [preview](https://dev-portal-git-product-integrations-hashicorp.vercel.app/waypoint/integrations/kubernetes?tab=platform))

```
./builtin
├── aws
│   ├── alb
│   │   ├── components 👈 (new folder)
│   │   │   ├── release-manager 👈
│   │   │   │   ├── parameters.hcl 👈
│   │   │   │   ├── outputs.hcl 👈
│   │   │   │   └── README.md 👈
```

These will eventually power the integrations frontend like so:

CleanShot 2022-12-16 at 09 50 54@2x

The alternative is manually writing ~100 files. Example:
- https://github.com/BrandonRomano/waypoint/pull/3

### How

This is largely modeled after the existing `mdxFormat` function but has a few changes to output up to 3 files alongside plugins within the `builtin/` directory. 

This leverages existing calls to [`waypoint-plugin-sdk/docs`](github.com/hashicorp/waypoint-plugin-sdk/docs).

###### Related

- https://github.com/hashicorp/waypoint/pull/151

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203295724413023

### CODE REVIEW

File: internal/cli/app_docs.go [hunk 1]
```diff
@@ -419,12 +421,235 @@ func (c *AppDocsCommand) mdxFormat(name, ct string, doc *docs.Documentation) {
...
+		// make component folder
+		os.MkdirAll(fmt.Sprintf("./builtin/%s/components/%s", pluginPath, componentSlug), os.ModePerm)
+
+		// populate README.md
+		readme, err := os.Create(fmt.Sprintf("./builtin/%s/components/%s/readme.md", pluginPath, componentSlug))
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Fprintln(readme, "<!-- This file was generated via `make gen/integrations-hcl -->")
```
Comment (krantzinator) [inline internal/cli/app_docs.go:510]: super-nit
```suggestion
		fmt.Fprintln(readme, "<!-- This file was generated via `make gen/integrations-hcl` -->")
```

**Review by krantzinator — APPROVED** [review]
🎉

**Comment by krantzinator** [comment]
We could also add this as a check in our CI @hashicorp/waypoint-ecosystem

**Review by demophoon — APPROVED** [review]
This is amazing

**Comment by thiskevinwang** [comment]
@krantzinator I think we hold on adding the CI check (https://github.com/hashicorp/waypoint/pull/4343) until integrations makes its way out of POC and into a more official beta state. We don't want to block any PRs quite yet nor cause any extra noise.

But it'd still be great to merge _this PR_ so we can pull this into our downstream integrations-fork (https://github.com/BrandonRomano/waypoint/tree/main). We may continue to iron out some kinks there.

What do you think?

**Comment by krantzinator** [comment]
@thiskevinwang sounds good!

**Pull request merged**